data/lua/core/unit_test.lua

```diff
@@ -0,0 +1,212 @@
+
+--[========[Unit Test Helpers]========]
+
+if rawget(_G, 'unit_test') ~= nil then
+	print("Loading unit_test module...")
+	
+	--! Causes the test to complete with a specific result.
+	--! This is mostly an internal detail but might very occasionally be useful to produce a result other than pass or fail
+	function unit_test.finish(result)
+		wesnoth.wml_actions.endlevel{
+			test_result = result,
+			linger_mode = true
+		}
+	end
+
+	--! End the test in unconditional failure
```

Comment:
I'm not actually 100% sure, but I think it will run events that are currently queued but will exit as soon as the event queue is empty. For example, if this is a "new turn" event, the "turn refresh" will still execute.

It definitely doesn't exit the current event.